### PR TITLE
Fix cache-related tests

### PR DESCRIPTION
**Pull request by zucchini-nlp**
# What does this PR do?

As per title, failing after latest [cache compatibility PR](https://github.com/huggingface/transformers/pull/38635)

### CODE REVIEW

**Comment by zucchini-nlp** [comment]
run-slow: kyutai_speech_to_text, musicgen_melody, qwen2_5_omni, rag, roformer, superglue

**Comment by zucchini-nlp** [comment]
run-slow: kyutai_speech_to_text, musicgen, musicgen_melody, rag, roformer, superglue

File: tests/pipelines/test_pipelines_image_text_to_text.py
Comment (ydshieh) [inline tests/pipelines/test_pipelines_image_text_to_text.py:1]: do you know what causse this changes?

File: tests/models/qwen2_5_vl/test_modeling_qwen2_5_vl.py [hunk 1]
```diff
@@ -403,7 +403,6 @@ def attention_mask_padding_matches_padding_free_with_position_ids(
                 logits_padded = res_padded.logits[inputs_dict["attention_mask"].bool()]
                 logits_padfree = res_padfree.logits[0]
 
-                torch.testing.assert_close(logits_padded.argmax(-1), logits_padfree.argmax(-1), rtol=0, atol=0)
```
Comment (ydshieh) [inline tests/models/qwen2_5_vl/test_modeling_qwen2_5_vl.py:406]: ok, can't find it on the common test file, so fine.

File: src/transformers/models/roformer/modeling_roformer.py [hunk 2]
```diff
@@ -261,6 +261,17 @@ def forward(
...
+            # Apply RoPE if self attention
+            if not is_cross_attention and sinusoidal_pos is not None:
+                if self.rotary_value:
+                    query_layer, key_layer, value_layer = self.apply_rotary_position_embeddings(
+                        sinusoidal_pos, query_layer, key_layer, value_layer
+                    )
+                else:
+                    query_layer, key_layer = self.apply_rotary_position_embeddings(
+                        sinusoidal_pos, query_layer, key_layer
+                    )
```
Comment (ydshieh) [inline src/transformers/models/roformer/modeling_roformer.py:273]: was it deleted at some point and here you just add it back?

**Review by ydshieh — APPROVED** [review]
Not very familiar with some changes in `src/transformers` so I wll rely on CI and trust you. Just a few nit questions to get some idea.

Thank you!

Could you link the PR you mentioned in this PR description?

File: src/transformers/models/roformer/modeling_roformer.py (hunk 2 above)
Comment (zucchini-nlp) [reply src/transformers/models/roformer/modeling_roformer.py:273]: yep, accidentally deleted RoPE 🙈

File: tests/models/qwen2_5_vl/test_modeling_qwen2_5_vl.py (hunk 1 above)
Comment (zucchini-nlp) [reply tests/models/qwen2_5_vl/test_modeling_qwen2_5_vl.py:406]: this one doesn't make sense because we want to check logits, not sampled argmax tokens. Even tiny diff in logits can give different tokens, and the next line check with `torch.allclose` is enough

File: tests/pipelines/test_pipelines_image_text_to_text.py
Comment (zucchini-nlp) [reply tests/pipelines/test_pipelines_image_text_to_text.py:1]: yep, after https://github.com/huggingface/transformers/pull/39374 we started using higher default max length. Text-only generation pipe already uses it, so it's fine
